appwrite/appwrite
# feat: X domain console cookie

## What does this PR do?

Adds an additional env var `_APP_CONSOLE_ROOT_SESSION`

When `_APP_CONSOLE_ROOT_SESSION=enabled`, console sessions will be stored on the root of the host domain.

e.g. 
by default, matching existing behavour `_APP_CONSOLE_ROOT_SESSION=disabled`, console sessions with have .cloud.appwrite.io cookie domain.
if the host domain is cloud.appwrite.io, and `_APP_CONSOLE_ROOT_SESSION=enabled` , console sessions with have .appwrite.io cookie domain.

## Test Plan

Manual testing: 
**NOTE** has to be https for cross domain cookies to work so accept the bad cert
Create a console account on https://x-domain-cookie-test.appwrite.org, where I have set env with `_APP_CONSOLE_ROOT_SESSION=enabled` and verify the cookie domain is .appwrite.org.

Create a non console / project account on https://x-domain-cookie-test.appwrite.org and verify the cookie domain is .x-domain-cookie-test.appwrite.org

## Checklist
